storage/interface.go

```diff
@@ -101,22 +95,19 @@ type ChunkQuerier interface {
 	Select(sortSeries bool, hints *SelectHints, matchers ...*labels.Matcher) ChunkSeriesSet
 }
 
-// Closer is io.Closer.
-type Closer interface {
-	// Close releases all resources required by queriers.
-	// This might be mean that other operations like head compaction or block release are
-	// blocked until this is closed.
-	Close() error
-}
-
-// LabelQuerier provides querying access over labels.
-type LabelQuerier interface {
+// LabelCloserQuerier provides querying access over labels.
+type LabelCloserQuerier interface {
```

Comment:
I don't think having Closer in here is useful, we don't call Blocks BlockCloser for example.

Reply:
Yes, but LabelQuerier it's extremely confusing. It's so easier to forget you have to close those

Reply:
It's more confusing to arbitrarily have Close in some interface names and not others. Without is cleaner - and half the time it's not as if the type is going to show up in the code anyway for you to notice that Close needs to be called.